saltproc/saltproc.py

```diff
@@ -163,37 +356,61 @@ def reopen_db(self, restart):
+        self.driver_refill_tank_db = self.f['driver refill tank composition']
+        self.driver_after_db = self.f['driver composition after reproc']
+
+        self.blanket_before_db = self.f['blanket composition before reproc']
+        self.blanket_after_db = self.f['blanket composition after reproc']
+        self.blanket_refill_tank_db = self.f['blanket refill tank composition']
+
+        self.waste_tank_db = self.f['waste tank composition']
+        self.isolib_db = self.f['iso names']
+        self.fissile_tank_db = self.f['fissile tank composition']
+        self.isozai_db = self.f['iso zai']
 
         if restart:
+            # resize dataset
+            self.isoname = [str(x) for x in self.isolib_db]
+            self.isozai = self.isozai_db
+
+            self.get_mat_def()
+            self.number_of_isotopes = len(self.isoname)
+
+            self.keff = self.keff_eoc_db[0, :]
             # set past time
             # !! this time thing should be made certain
             self.current_step = np.amax(np.nonzero(self.keff)) + 1
-
+            self.current_step = np.amax(np.nonzero(sum(self.driver_before_db))) + 1
             # resize datasets
-            self.keff_db.resize((2, self.steps + self.current_step))
-            self.keff_db_0.resize((2, self.steps + self.current_step))
+            self.keff_eoc_db.resize((2, self.steps + self.current_step))
+            self.keff_boc_db.resize((2, self.steps + self.current_step))
             shape = (self.steps + self.current_step +
                      1, self.number_of_isotopes)
-            self.bu_adens_db_0.resize(shape)
-            self.bu_adens_db_1.resize(shape)
-            self.tank_adens_db.resize(shape)
-            self.noble_adens_db.resize(shape)
-            self.th_adens_db.resize(shape)
-            self.rem_adens = np.zeros((5, self.number_of_isotopes))
-            self.th232_adens_0 = self.bu_adens_db_0[0, self.th232_id]
+            self.driver_before_db.resize(shape)
+            self.driver_after_db.resize(shape)
+            self.driver_refill_tank_db.resize(shape)
+
+            self.blanket_before_db.resize(shape)
+            self.blanket_after_db.resize(shape)
+            self.blanket_refill_tank_db.resize(shape)
+
+            self.waste_tank_db.resize(shape)
+            self.fissile_tank_db.resize(shape)
+
+            # write new material file
+            self.core = {}
+            self.core[self.driver_mat_name] = self.driver_after_db[self.current_step - 2]
+            self.core[self.blanket_mat_name] = self.blanket_after_db[self.current_step - 2]
+            print(self.current_step - 1)
+            print(self.core[self.driver_mat_name])
+            print(self.core[self.blanket_mat_name])
+            self.write_mat_file()
 
     def read_res(self, moment):
-        """ Reads using PyNE the SERPENT output .res file   
+        """ Reads SERPENT output .res file   
```

Comment:
why not use pyne any more? it looks like it takes a lot of messy code to replace the pyne capabilities that were used before.